### PR TITLE
There, I fixed issue #173

### PR DESCRIPTION
**Pull request by bodocabdd**
This should do the trick. Not sure if this is the most elegant solution, but worked for me.
Hope it helps. ;)

Fixes issue #173

### CODE REVIEW

File: lib/error-matcher.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ module.exports = (function () {
 
     var output = '';
     var lastEnd = 0;
-    for (var matchIndex in this.currentMatch) {
+    for (var matchIndex = 0; this.currentMatch.length && matchIndex <= this.currentMatch.length; matchIndex++) {
```
Comment (noseglid) [inline lib/error-matcher.js]: This doesn't seem right. The index could never be equal to the length of the array.

**Comment by noseglid** [comment]
I don't see how this code differs from what was there before?

File: lib/error-matcher.js (hunk 1 above)
Comment (bodocabdd) [inline lib/error-matcher.js]: Sorry. It was supposed to be < only, not <=. My bad. Didn't catch that.

**Comment by bodocabdd** [comment]
It's different. You use ```for(var index in object)``` with array, and it was detecting "unique" proto function as enumerable property, which was triggering the error.

**Comment by bodocabdd** [comment]
I guess you could use a ```for(var index of array)``` or ```array.forEach``` also.

**Comment by noseglid** [comment]
I can see how this would be a problem if `this.regex` contained anything else than regular expressions. But what I don't get is how it can contain other stuff for you? It seems as there is something more happening here.

**Comment by bodocabdd** [comment]
Hmm... I looked up my packages. Seems that the "atom-permute" deprecated package is causing the problem. "unique" is declared as a prototype function for Array. That explains why it is detected as enumerable property when you use ```for (key in array)```. Disabling the package solves the problem. But I still recommend you change those lines...

**Comment by noseglid** [comment]
Ohh extending the prototype! It's forbidden! Forbidden I say! Uninstall that crap and throw it away :)

I've implemented a more defensive strategy in #177 . Would you mind testing if that works for you?

**Comment by bodocabdd** [comment]
Already did, right after seeing that. ;) Never used atom-permute anyway.
Your strategy is very good, sir! ES6 FTW! I guess this closes the case.

Thank you very much for your time.

**Comment by noseglid** [comment]
Thank you for taking the time to report and debugging this with me!

**Pull request closed without merging**